### PR TITLE
VACMS-6177: Add moderation state GraphQL plugin.

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Relates to #6177.  Acting on @indytechcook's advice, I did the simplest possible thing and added a field plugin.

## Testing done
I'm gonna let the automated tests run and see if they turn up anything.  Meanwhile I'll continue playing in GraphQL Explorer and see if I turn up any dealbreakers.

## Screenshots
Screen Shot 2021-09-01 at 10 38 24 AM
Screen Shot 2021-09-01 at 10 38 40 AM
Screen Shot 2021-09-01 at 10 39 00 AM

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [ ] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_graphql/src/Plugin/GraphQL/Fields/ModerationState.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * )
+ */
+class ModerationState extends FieldPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function resolveValues($value, array $args, ResolveContext $context, ResolveInfo $info) {
+    if ($value instanceof FieldableEntityInterface) {
+      if ($value->hasField('moderation_state')) {
```
Comment (indytechcook) [inline docroot/modules/custom/va_gov_graphql/src/Plugin/GraphQL/Fields/ModerationState.php:30]:
```suggestion
    if ($value instanceof FieldableEntityInterface && $value->hasField('moderation_state')) {
```

Comment (indytechcook) [reply docroot/modules/custom/va_gov_graphql/src/Plugin/GraphQL/Fields/ModerationState.php:30]: An remove the other `}`

**Review by indytechcook — DISMISSED** [review]

**Comment by indytechcook** [comment]
I do think it's strange that the left side shows the red underline under the field.

**Comment by ndouglas** [comment]
I hard-refreshed the page and that went away, must've been a browser-side cache thing.

**Comment by ndouglas** [comment]
@indytechcook I had a brain fart and removed the patch, then realized and re-added it -- all function LGTM now.

Screen Shot 2021-09-01 at 2 10 43 PM

**Comment by ndouglas** [comment]
Fundamentally not crazy about that patch, but I don't think anyone is 😂

**Review by indytechcook — APPROVED** [review]

**Pull request merged**